languages/api.py

```diff
@@ -0,0 +1,107 @@
+import langcodes
+import hashlib
+import click
+import json
+
+from flask import Blueprint, request, jsonify, cli
+
+from .models import db, LanguagePack, Language
+from .settings import config
+from .s3 import upload_pbl
+from .utils import id_generator
+
+parent_app = None
+api = Blueprint('api', __name__)
+
+@api.route('/languages')
+def get_languages():
+    packs = LanguagePack.query
+    hardware = request.args.get('hw')
+    if hardware:
+        packs = packs.filter_by(hardware=hardware)
+    firmware = request.args.get('firmware')
+    if firmware:
+        firmware_major, firmware_minor = [int(part) for part in firmware.split('.')[:2]]
+        packs = packs.filter(LanguagePack.firmware_major == firmware_major, LanguagePack.firmware_minor <= firmware_minor)
+    return jsonify({
+        'languages': [pack.to_json() for pack in packs]
+    })
+
+
+@api.route('/languages/<id>')
+def language(id):
+    pack = LanguagePack.query.filter_by(id=id).one_or_none()
+    if pack is None:
+        return api_error(404)
+    return jsonify(pack.to_json())
+
+
+@api.route('/languages/submit', methods=['POST'])
+def submit_language_pack():
+    secret_key = ''
+    header = request.headers.get('Authorization')
+    if header:
+        auth = header.split(' ')
+        if len(auth) == 2 and auth[0] == 'Bearer':
+            secret_key = auth[1]
+    if secret_key != config['SECRET_KEY']:
+        return 'Not authorized', 410
```

Comment:
this is traditionally not we what we use `SECRET_KEY` for -- how about having this be a click command instead that gets run locally on someone's machine, for now?